src/dns.c

```diff
@@ -2403,30 +2430,36 @@ hsk_dns_name_alloc(
 }
 
 bool
-hsk_dns_name_dirty(const char *name) {
-  char *s = (char *)name;
+hsk_dns_verify_name(const char *name) {
```

Comment:
What's the difference between hsk_dns_verify_name hsk_dns_name_verify ?
This function seems simpler.

Reply:
ugh you're right there's a name collision there. The new function is more restrictive like `rules.verifyName()` in hsd, which doesn't allow special characters or byte-codes at all. By filtering out names with slashes and responding with REFUSED we can make the NSEC proof logic simpler, ie we know the only `\000` and `\255` stuff in these names will be from internal functions.

The older more complicated function is more flexible and more to spec with DNS which allows non-printable byte codes and that logic was also opened up for these NSEC proofs. Maybe I'll change the name of this one back to "dirty"